lib/rubocop/cop/naming/predicate_name.rb

```diff
@@ -121,6 +150,15 @@ def validate_config
 
         private
 
+        def boolean_sorbet_sig?(node)
+          # Select if it's a `sig` block with a `returns(T::Boolean)` directly above the method.
+          node.parent.each_descendant(:block).any? do |block_node|
+            block_node.method?(:sig) &&
+              block_node.source.include?('returns(T::Boolean)') &&
+              block_node.loc.end.line == node.loc.line - 1
+          end
```

Comment:
This is likely brittle. For one thing, it seems that there is allowed to be whitespace between `sig` and `def`, which would cause this to not return a node. Although `rubocop-ast` discourages `left_sibling`, I think it's probably the most straightforward way to find the `sig` node.

```suggestion
          return false unless (prev_node = node.left_sibling)

          prev_node.method?(:sig) &&
            prev_node.source.include?('returns(T::Boolean)')
```

I also have a concern about hardcoding the source to check. If nothing else, whitespace in the signature will cause this to not match (eg. `returns( T::Boolean )`). I'd prefer to use a pattern matcher for this check.

Please also add tests for blank lines between `sig` and `def`.

Reply:
If this is a pattern we're going to potentially adopt elsewhere, I think this method is a good candidate for making a bit more generic so that the return type can be passed in instead of hardcoded.

Reply:
> I'd prefer to use a pattern matcher for this check.

I tried that but I couldn’t figure out the syntax - any tips? I can see that `sig { returns(T::Boolean) }` has an AST representation of

```ruby
(block\n  (send nil :sig)\n  (args)\n  (send nil :returns\n    (const\n      (const nil :T) :Boolean)))
```

But `sig { returns(T::Array[String]) }` has a different number of `const`, like so:

```ruby
(block\n  (send nil :sig)\n  (args)\n  (send nil :returns\n    (send\n      (const\n        (const nil :T) :Array) :[]\n      (const nil :String))))

```

And `sig { returns(String) }` is different again.

```ruby
(block\n  (send nil :sig)\n  (args)\n  (send nil :returns\n    (const nil :String)))
```

How do I get just the bit after the `returns` send node - whatever the structure is? So that I can check that return type, from the pattern matcher, is a T::Boolean or whatever?

Reply:
I did it a less sophisticated way in the interim. That method is now slightly more generic and copes with more variants, spacing, etc.

Reply:
I'll add a suggestion.